Inspector/validate.h

```diff
@@ -0,0 +1,12 @@
+#ifndef validate_h
+#define validate_h
+
+#include <stdio.h>
+#include <stdbool.h>
+
+#include "cs_magic.h"
+
+
+bool validate(const void *data, size_t length, size_t slot, const CS_CodeDirectory *cd);
```

Comment:
I don't know, to me "validate" sounds like a command, not a query. But then again, in C I'm not sure what a "validate" command would do. In C++ I'd expect it to return void or throw an exception. I'd probably call it "is_valid", but maybe it doesn't matter in C.